src/common.c

```diff
@@ -315,7 +320,26 @@ const char *ofi_straddr(char *buf, size_t *len,
 				str, *((uint16_t *)addr + 8));
 		break;
 	case FI_SOCKADDR_IB:
-		size = snprintf(buf, *len, "fi_sockaddr_ib://%p", addr);
+#if HAVE_RDMA_RDMA_CMA_H
+		sib = addr;
+		sib_addr = &sib->sib_addr;
+
+		assert(sizeof(str) >= 40);
+		if (sprintf(str, "%04x:%04x:%04x:%04x:%04x:%04x:%04x:%04x",
+		            ntohs(sib_addr->sib_addr16[0]),
+		            ntohs(sib_addr->sib_addr16[2]),
+		            ntohs(sib_addr->sib_addr16[3]),
+		            ntohs(sib_addr->sib_addr16[4]),
+		            ntohs(sib_addr->sib_addr16[5]),
+		            ntohs(sib_addr->sib_addr16[6]),
+		            ntohs(sib_addr->sib_addr16[7]),
+		            ntohs(sib_addr->sib_addr16[8])) != 39)
+			return NULL;
+
+		size = snprintf(buf, *len, "fi_sockaddr_ib://%s", str);
+#else
+		return NULL;
+#endif
```

Comment:
Treat GID as ipv6 (use inet_ntop) and avoid the #if and adding verbs provider specific code into the common core.

Reply:
@sydidelot  Check out how we implement FI_ADDR_EFA, which does exactly that. EFA does not support IP addressing and directly works witih device GIDs.

Reply:
@shefty @rajachan Thanks for your help! I checked out the EFA provider and I see that EFA declares its own address format that contains the GID. I can modify the code so an IB address corresponds to a GID and use `inet_ntop` to print it.
If we go that direction, I would suggest renaming `FI_SOCKADDR_IB` because technically, the address won't be a IB `sockaddr` (`struct sockaddr_ib`) address anymore. Do you guys have any thoughts on it?